### ISSUE TITLE
Command line argument handling does not work in release builds

### ISSUE THREAD

**Issue by SaschaWillems** [KhronosGroup/Vulkan-Samples#1249]
When running a release build, all command line arguments that require an additional argument fail like this:

![image](https://github.com/user-attachments/assets/2af88f43-46d9-4136-b826-e214fb0ebbfa)

Command line was:

sample mesh_shading --shading-language hlsl

In a debug build, command line argument handling works fine. This hints at undefined behavior somewhere in the argument handling of the framework.

**Comment by TzuChieh** [comment]
Reversing the arguments will work though, for example

> --shading-language hlsl sample mesh_shading

I am encountering the same issue with Visual Studio 2022 under Windows 10.

**Comment by SaschaWillems** [comment]
This is nigh impossible to debug. The issue only shows up in a proper release build. Doing a "release with debug" build in MSVC works fine. I'm looking it into, but I'm kinda stumped. This is sadly a part of our framework with close to no code documentation.

**Comment by gpx1000** [comment]
I'll take a look if I get a chance; but it seems to work fine in Linux (as expected).  Windows doesn't like things to perform the exact same in debug and release, so it's one of those bugs...  I might not be able to help until the next time I'm in front of a Windows machine.

**Comment by SaschaWillems** [comment]
I'm trying to get behind the issue on Windows. If I can't find the reason I'll scream for help ;)

When setting the C++ warning level to all, I actually do get some warnings from MSVC that hint at an issue with the use of auto. Might be a clue.

**Comment by gpx1000** [comment]
Understood, I'll hold off looking at it then.  Lemme know if you need help.

**Comment by SaschaWillems** [comment]
The deeper I dig into this, the more I feel like the whole CLI parser in our samples does something very wrong (at least when compiled with MSVC):

![image](https://github.com/user-attachments/assets/f1dd4110-eb8e-4375-99be-072a19fe5378)

**Comment by gpx1000** [comment]
;-) I keep pushing to have warnings on highest level on all platforms and treated as errors.   Warnings are how the IDE does it's best to help us and ignoring the help leads to bad days.

**Comment by SaschaWillems** [comment]
Yeah, we probably should discuss that on one of our next calls. There are lots of warnings in our code when all warnings are enabled. The CLI11 library we use for CLI command parsing also has a slew of compiler warnings, maybe it's time to replace it. The code for command line argument handling in our samples seems unnecessarily complex.

**Comment by nilspin** [comment]
Just wanted to confirm this bug only occurs on Release; RelWithDebInfo also parses cmd args correctly.
On Android 33, 34 too the Release build parses correctly. It seems the bug is only on MSVC.

**Comment by marty-johnson59** [comment]
Raising this for @tomadamatkinson 's attention.  Tom, this might be something you have insights into?
